ksqldb-common/src/main/java/io/confluent/ksql/name/ColumnNames.java

```diff
@@ -61,4 +70,42 @@ public static ColumnName generatedJoinColumnAlias(
   public static boolean isAggregate(final ColumnName name) {
     return name.text().startsWith(AGGREGATE_COLUMN_PREFIX);
   }
+
+  /**
+   * Determines the next unique column alias.
+   *
+   * <p>Finds any existing {@code KSQL_COL_x} column names in the supplied {@code sourceSchema} to
+   * ensure the returned generated column name is unique.
+   *
+   * @param sourceSchema the source schema.
+   * @return a column name in the form {@code KSQL_COL_x} which does not clash with source schema.
+   */
+  public static ColumnName nextGeneratedColumnAlias(final LogicalSchema sourceSchema) {
```

Comment:
I feel like we should generate the name of the partition by column during the `AstSanitizer` step (just as we do for other expressions without column names). Unfortunately that probably requires supporting column names in `PARTITION BY` - is that something that's on your radar? This isn't a blocker for this PR, but I want to make sure we have a path forward that's backwards compatible.

This change might be hard to implement in the future - we generate `KSQL_COL_x` based on the location of the column, not the number of previous generated columns, so the algorithms wouldn't necessarily line up. Is this `nextGeneratedColumnAlias` specified anywhere in the serialized plan?

Reply:
Aliasing in PARTITION BY is definitely on my radar: https://github.com/confluentinc/ksql/issues/4813.  I was hoping you may be able to pick that one up!

And yes, once this is done we can make this call from `AstSanitizer`

> This change might be hard to implement in the future - we generate KSQL_COL_x based on the location of the column, not the number of previous generated columns, so the algorithms wouldn't necessarily line up.

With https://github.com/confluentinc/ksql/pull/4827 we actually generate the index based on other column names. 

>  Is this nextGeneratedColumnAlias specified anywhere in the serialized plan?

The sink schema is in the serialized plan.